py/server/deephaven/column.py

```diff
@@ -202,15 +206,30 @@ def datetime_col(name: str, data: Sequence) -> InputColumn:
     Returns:
         a new input column
     """
-    data = [time.to_j_instant(d) for d in data]
+
+    # try to convert to numpy array of datetime64 if not already, so that we can call translateArrayLongToInstant on
+    # it to reduce the number of round trips to the JVM
+    if not isinstance(data, np.ndarray):
+        try:
+            data = np.array([pd.Timestamp(dt).to_numpy() for dt in data], dtype=np.datetime64)
+        except Exception as e:
+            ...
+
+    if isinstance(data, np.ndarray) and data.dtype.kind == 'M':
+        longs = jpy.array('long', data.astype('datetime64[ns]').astype('int64'))
+        data = _JPrimitiveArrayConversionUtility.translateArrayLongToInstant(longs)
```

Comment:
Are there concerns about this pattern doubling the memory allocations?

Reply:
I think it is not so bad. IMO the performance gain outweighs the additional memory consumption.